source/Octopus.Client/Repositories/Async/BasicRepository.cs

```diff
@@ -37,25 +37,35 @@ protected BasicRepository(IOctopusAsyncRepository repository, string collectionL
         public IOctopusAsyncClient Client { get; }
         public IOctopusAsyncRepository Repository { get; }
 
-        protected virtual void CheckSpaceResource(IHaveSpaceResource spaceResource)
+        protected virtual async Task CheckSpaceResource(IHaveSpaceResource spaceResource)
         {
-            Repository.Scope.Apply(
+            await Repository.Scope.Apply(
```

Comment:
[nit] You could just return the promise here instead

`return Repository.Scope.Apply(...)` + the method does not need to be `async`

This has the advantage of not needing the `.ConfigureAwait(false)`, because you have not done an `await`!